### PR TITLE
Create stock model and its tests

### PR DESCRIPTION
**Pull request by Rubncal04**
Create the stock model and its tests when we create the stocks for products

### CODE REVIEW

File: spec/models/stock_spec.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+require 'rails_helper'
+
+RSpec.describe Stock, type: :model do
+  it "can't be saved without a product" do
+    stock = Stock.new
+    stock.product = nil
+    expect { stock.save!(validate: false) }.to raise_error ActiveRecord::NotNullViolation
```
Comment (gabo-cs) [inline spec/models/stock_spec.rb:7]: Two questions here:
1) Why are you doing `validate: false`? What validations you want to skip?
2) You're validating that a stock shouldn't be saved without a product, and you're asserting against `ActiveRecord::NotNullViolation`. Shouldn't you be asserting against `ActiveRecord::RecordInvalid` instead. 

Doing `stock.save!` without a product should give you:
```
TRANSACTION (0.3ms)  ROLLBACK
ActiveRecord::RecordInvalid: Validation failed: Product must exist
```

File: spec/models/stock_spec.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  end
+
+  it "is valid with product" do
+    product = Product.create(name: "Cheese", code: "123ght")
+    expect(Stock.new(quantity: 2, product_id: product.id)).to be_valid
+  end
+
+  it "is valid with differents product id" do
```
Comment (gabo-cs) [inline spec/models/stock_spec.rb:15]:
```suggestion
  it "is valid with different product IDs" do
```

File: spec/models/stock_spec.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    product1 = Product.create(name: "Cheese", code: "123ght")
+    product2 = Product.create(name: "Cheese", code: "12ght")
+    stock1 = Stock.new(quantity: 2, product_id: product1.id)
+    stock2 = Stock.new(quantity: 3, product_id: product2.id)
+    expect(stock1 != stock2).to be_truthy
+  end
+
+  it "is not valid with product id equals" do
```
Comment (gabo-cs) [inline spec/models/stock_spec.rb:23]:
```suggestion
  it "is not valid with product equal IDs" do
```

**Review by gabo-cs — CHANGES_REQUESTED** [review]
Another question: Are products going to have multiple stocks? I don't see the `has_one`/`has_many` line change in the `Product` model. Can you give me an example of how that relationship going to work in your business logic?

File: spec/models/stock_spec.rb (hunk 1 above)
Comment (Rubncal04) [reply spec/models/stock_spec.rb:7]: No, the products are going to have one stock and that stock can be changed

Comment (Rubncal04) [reply spec/models/stock_spec.rb:7]: respect of your questions, the idea is validate that `product_id` exists and it cannot be created without it

**Comment by Rubncal04** [comment]
Implement another fix

**Pull request closed without merging**